CRM/Cdntaxcalculator/BAO/CDNTaxes.php

```diff
@@ -1,58 +1,362 @@
+    $result = civicrm_api3('Event', 'get', [
+      'id' => $event_id,
+      'return.loc_block_id' => 1,
+      'api.LocBlock.get' => [
+        'api.Address.get' => [],
+      ],
+    ]);
+
+    foreach ($result['values'] as $key => $val) {
+      if (isset($val['api.LocBlock.get'])) {
+        foreach ($val['api.LocBlock.get']['values'] as $loc) {
+          if (isset($loc['api.Address.get'])) {
+            foreach ($loc['api.Address.get']['values'] as $addr) {
+              if (!empty($addr['state_province_id'])) {
+                $province_id = $addr['state_province_id'];
+                $country_id = $addr['country_id'];
+              }
+            }
+          }
+        }
+      }
+    }
+
+    if (empty($province_id)) {
+      $domain_id = CRM_Core_Config::domainID();
+
+      $result = civicrm_api3('Domain', 'getsingle', [
+        'id' => $domainID,
+      ]);
+
+      if (empty($result['domain_address']['state_province_id'])) {
+        CRM_Core_Error::fatal("The state/province of the default domain is not set (Administer > Communications > Organisation address)");
+      }
+
+      $province_id = $result['domain_address']['state_province_id'];
+      $country_id = $result['domain_address']['country_id'];
+    }
+
+    if (empty($province_id) || empty($country_id)) {
+      CRM_Core_Error::fatal("Failed to find a default country/province for the event.");
+    }
+
+    if ($country_id == 1039) {
+      $taxes = $cdnTaxes[$province_id];
+      $taxes['TAX_TOTAL'] = $taxes['HST_GST'] + $taxes['PST'];
+      $taxes['province_id'] = $province_id;
+    }
+
+    return $taxes;
+  }
+
+  /**
+   * Checks to see if there are any taxable amounts in the priceset.
+   */
+  static function hasTaxableAmounts($feeBlock) {
+    foreach ($feeBlock as $fee) {
+      if (!is_array($fee['options'])) {
+        continue;
+      }
+
+      foreach ($fee['options'] as &$option) {
+        if (!empty($option['tax_rate'])) {
+          return TRUE;
+        }
       }
     }
-    if ($state) {
-      return CRM_Utils_Array::value($state, self::$_totalTax);
+
+    return FALSE;
+  }
+
+  /**
+   * Checks whether the FT associated with an event has taxes.
+   *
+   * NB: an event priceset could include items linked to various FT,
+   * some taxable and some not, but the event itself is associated to
+   */
+  static function isEventFinancialTypeTaxable($event_id) {
+    $result = civicrm_api3('Event', 'getsingle', [
+      'event_id' => $event_id,
+      'return.financial_type_id' => 1,
+    ]);
+
+    $financial_type = $result['financial_type_id'];
+
+    $is_taxed = CRM_Core_DAO::singleValueQuery('
+      SELECT count(*)
+        FROM civicrm_entity_financial_account efa
+        LEFT JOIN civicrm_financial_account fa on fa.id = efa.financial_account_id
+       WHERE efa.entity_id = 9
+         AND efa.entity_table = "civicrm_financial_type"
+         AND fa.is_tax = 1', [
+      1 => [$event_id, 'Positive'],
+    ]);
+
```

Comment:
This is not always true since an event might be configured to have priceset having some price fields taxable. So it might be possible that event.financial_type_id is not taxable.